openbas-api/src/test/java/io/openbas/rest/ExerciseLessonsApiTest.java

```diff
@@ -0,0 +1,133 @@
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.time.Instant;
+import java.util.List;
+
+import static io.openbas.database.model.ExerciseStatus.SCHEDULED;
+import static io.openbas.rest.exercise.ExerciseApi.EXERCISE_URI;
+import static io.openbas.utils.JsonUtils.asJsonString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.mockito.Mockito.*;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ExerciseLessonsApiTest extends IntegrationTest {
+
+  static Exercise EXERCISE;
+  static LessonsCategory LESSONCATEGORY;
+  static Team TEAM;
+  static User USER;
+
+  @Autowired
+  private MockMvc mvc;
+  @Autowired
+  private ExerciseService exerciseService;
+  @Autowired
+  private ExerciseRepository exerciseRepository;
+  @Autowired
+  private LessonsCategoryRepository lessonsCategoryRepository;
+  @SpyBean
+  private MailingService mailingService;
+  @Autowired
+  private TeamRepository teamRepository;
+  @Autowired
+  private UserRepository userRepository;
+
+  @BeforeAll
+  void beforeAll() {
+    EXERCISE = getExercise();
+    LESSONCATEGORY = getLessonCategory();
+  }
+
+  @AfterAll
+  void afterAll() {
+    this.exerciseRepository.delete(EXERCISE);
+    this.lessonsCategoryRepository.delete(LESSONCATEGORY);
+    this.teamRepository.delete(TEAM);
+    this.userRepository.delete(USER);
+  }
+
+
+  private LessonsCategory getLessonCategory() {
+    LessonsCategory lessonsCategory = new LessonsCategory();
+    lessonsCategory.setExercise(EXERCISE);
+    lessonsCategory.setName("Category");
+    lessonsCategory.setDescription("Description");
+    lessonsCategory.setOrder(0);
+    lessonsCategory.setTeams(List.of(getTeam()));
+    return this.lessonsCategoryRepository.save(lessonsCategory);
+  }
+
+  private Team getTeam() {
+    Team team = new Team();
+    team.setName("My team");
+
+    User user = new User();
+    user.setEmail("testSurvey@gmail.com");
+    USER = this.userRepository.save(user);
+
+    team.setUsers(List.of(USER));
+    TEAM = this.teamRepository.save(team);
+    return TEAM;
+  }
+
+  private Exercise getExercise() {
+    Exercise exercise = new Exercise();
+    exercise.setName("Exercice name");
+    exercise.setStatus(SCHEDULED);
+    exercise.setFrom("test@test.com");
+    exercise.setReplyTos(List.of("test@test.com"));
+    exercise.setStart(Instant.now());
+    return this.exerciseService.createExercise(exercise);
+  }
+
+  @DisplayName("Send surveys for exercise lessons")
+  @Test
+  @Order(1)
```

Comment:
this annotation is unnecessary in this file since it contains only one testing function, moreover unit testing function should be independant so they should be processed at any order